pkg/config/analysis/local/context.go

```diff
@@ -93,7 +100,11 @@ func (i *istiodContext) GetMessages(analyzerNames ...string) diag.Messages {
 
 func (i *istiodContext) Find(col config.GroupVersionKind, name resource.FullName) *resource.Instance {
 	i.collectionReporter(col)
-	k := key{col, name, "default"}
+	k := key{
```

Comment:
Can this be consistent with the key you modified in `ForEach`? Otherwise it's not using the cache. BTW if you use UID, do we still need to include all other fields as the key?

Reply:
Thanks for flagging this. Missed changing this. Updated now.

> BTW if you use UID, do we still need to include all other fields as the key?

I think those are needed as well. I didn't dig deep but removing the other fields causes the unit tests to fail